src/lex/cpp.rs

```diff
@@ -210,7 +210,11 @@ impl<'a> PreProcessor<'a> {
     }
     #[inline]
     fn next_token(&mut self) -> Option<CppResult<Token>> {
-        self.lexer_mut().next()
+        match self.lexer_mut().next() {
+            Some(Err(err)) => Some(Err(err.into())),
+            Some(Ok(ok)) => Some(Ok(ok)),
+            None => None,
```

Comment:
This could be replaced be some `map`s, do you prefer the `match` for some reason?